components/browser/engine-gecko-nightly/src/test/java/mozilla/components/browser/engine/gecko/media/GeckoMediaTest.kt

```diff
@@ -28,36 +28,47 @@ class GeckoMediaTest {
 
         delegate.onPlaybackStateChange(mediaElement, MediaElement.MEDIA_STATE_PLAYING)
         assertEquals(Media.PlaybackState.PLAYING, media.playbackState)
+        assertEquals(Media.State.PLAYING, media.state)
 
         delegate.onPlaybackStateChange(mediaElement, MediaElement.MEDIA_STATE_SEEKING)
         assertEquals(Media.PlaybackState.SEEKING, media.playbackState)
+        assertEquals(Media.State.PLAYING, media.state)
 
         delegate.onPlaybackStateChange(mediaElement, MediaElement.MEDIA_STATE_WAITING)
         assertEquals(Media.PlaybackState.WAITING, media.playbackState)
+        assertEquals(Media.State.PLAYING, media.state)
```

Comment:
This confused me a bit 🤔 , why when `media.playbackState`  is `waiting` `media.state` is `PLAYING`.

Reply:
Because it does not affect the already existing "playing" state. The "waiting" event means it's waiting for e.g. a seek to complete. But this doesn't mean it was paused. The media will still continue to play as soon as "waiting" is completed.

Reply:
Unfortunately `playbackState` mixes events and "states". And `Media.state` is me trying to dumb this down to a simple PLAYING, PAUSED, ENDED state.

Reply:
Got it thanks :)